src/scripts/main.js

```diff
@@ -1,3 +1,345 @@
+    }
+  }
+  cellSound.play();
+}
+
+function moveRow(movement) {
+  for (let i = 0; i < 16; i += 4) {
+    const row = cellsNumbers.slice(i, i + 4);
+    let filteredRow = row.filter(num => num > 0);
+
+    for (let index = 0; index < filteredRow.length; index++) {
+      if (filteredRow[index] === filteredRow[index + 1]) {
+        initialScore += +filteredRow[index + 1] * 2;
+        filteredRow[index + 1] *= 2;
+        filteredRow[index] = 0;
+        index++;
+      }
+    }
+
+    filteredRow = filteredRow.filter(num => num > 0);
+
+    const emptyCellsLength = 4 - filteredRow.length;
+    const zeroArray = Array(emptyCellsLength).fill(0);
+    let updatedRow;
+
+    if (movement === 'right') {
+      updatedRow = zeroArray.concat(filteredRow);
+    } else {
+      updatedRow = filteredRow.concat(zeroArray);
+    }
+
+    for (let j = 0; j < 4; j++) {
+      cellsNumbers[i + j] = updatedRow[j];
+    }
+  }
+  cellSound.play();
+}
+
+function resetGame() {
+  // Clear cellsNumbers array
+  cellsNumbers.fill(0);
+
+  // Reset cell HTML and class
+  cellsNodes.forEach(cell => {
+    cell.innerHTML = '';
+    cell.classList.remove(...Array.from(cell.classList));
+    cell.classList.add('field-cell');
+  });
+
+  // Reset score
+  initialScore = 0;
+  score.innerHTML = '0';
+
+  // Reset any other necessary variables or game state
+}
+
+function isWin() {
+  if (cellsNumbers.find(num => num === 2048)) {
+    disableKeyboardEvents();
+    winSound.play();
+
+    return true;
+  }
+}
+
+function isGameOver() {
+  if (cellsNumbers.includes(0)) {
+    return false;
+  }
+
+  // horizontal check
+
+  for (let row = 0; row < 4; row++) {
+    for (let col = 0; col < 3; col++) {
+      const index = row * 4 + col;
+
+      if (cellsNumbers[index] === cellsNumbers[index + 1]) {
+        return false;
+      }
+    }
+  }
+
+  // vertical  check
+
+  for (let row = 0; row < 3; row++) {
+    for (let col = 0; col < 4; col++) {
+      const index = row * 4 + col;
+
+      if (cellsNumbers[index] === cellsNumbers[index + 4]) {
+        return false;
+      }
+    }
+  }
+  disableKeyboardEvents();
+
+  return true;
+}
+
+startButton.addEventListener('click', () => {
+
+  if (isGameOver()) {
+    disableKeyboardEvents();
+    messageLose.classList.add('hidden');
+    messageStart.classList.add('hidden');
+    messageContainer.style.opacity = 0;
+    resetGame();
+  }
+  startButton.innerHTML = 'Start';
+  startButton.classList.add('restart');
+  startButton.innerHTML = 'Restart';
+  startButton.classList.remove('restart-lose');
+  resetGame();
+
+  getRandomCell(cellsNodes, cellsNumbers);
+  getRandomCell(cellsNodes, cellsNumbers);
+  updateBoard();
+  // score.innerHTML = sumScore();
+  enableKeyboardEvents();
+});
+
+function handleKeyDown(e) {
+  if (e.defaultPrevented) {
+    return; // Do nothing if the event was already processed
+  }
+
+  switch (e.key) {
+    case 'Down': // IE/Edge specific value
```

Comment:
not needed to comment your code, 
good code = code that don't need to be commented :)